### PR TITLE
[ORCT-197] amend eslint misuse

### PR DESCRIPTION
**Pull request by xsalonx**
#### I have a JIRA ticket
- [x] branch and/or PR name(s) include(s) JIRA ID
- [ ] issue has "Fix version" assigned
- [x] issue "Status" is set to "In review"
- [x] PR labels are selected

Notable changes for users:
- NA

Notable changes for developers (99% of changes if related only to eslint styling):
- by accident, directory `lib/database` was provided inside exclude clause of eslint configuration, what resulted in many hidden, styling errors
- make sequelize logging configurable (see `lib/database/DatabaseManager.js` and additions of var `RCT_DB_LOGGING`)
- remove problematic, unused, legacy code from `DatabaseService`

Changes made to the database:
- NA

### CODE REVIEW

**Review by martinboulais — CHANGES_REQUESTED** [review]

File: app/lib/database/utilities/PGQueryBuilder.js [hunk 1]
```diff
@@ -53,60 +54,52 @@ const handleBetween = (fieldName, pairsList) => {
...
-            exclude: [],
-            between: []
-        }
-
         // Mapping search params to categorized { key, value } pairs
-        const filterTypesRegex= new RegExp(Object.keys(filterTypes).map((t) => `(.*-${t})`).join('|'));
+        const filterTypesRegex = new RegExp(Object.keys(filterTypes).map((t) => `(.*-${t})`).join('|'));
         const filterParams = Object.entries(params)
-            .filter(([k, v]) => k.match(filterTypesRegex))
+            .filter(([k, _]) => k.match(filterTypesRegex))
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply app/lib/database/utilities/PGQueryBuilder.js:82]: indeed

File: app/lib/database/views/flags_view.js [hunk 2]
```diff
@@ -37,14 +37,14 @@ const dataFieldToSQLBelongsLogicalClause = (data, fieldName, isString) => {
...
 const flags_view = (query) => {
-    run_selection_sql = dataFieldToSQLBelongsLogicalClause(query.run_numbers, 'r.run_number')
-    detector_selection_sql = dataFieldToSQLBelongsLogicalClause(query.detector, 'ds.name', true)
+    const run_selection_sql = dataFieldToSQLBelongsLogicalClause(query.run_numbers, 'r.run_number');
+    const detector_selection_sql = dataFieldToSQLBelongsLogicalClause(query.detector, 'ds.name', true);
 
     const data_pass_sql = `dp.name = '${query.data_pass_name}'`;
     const whereClause = [data_pass_sql, run_selection_sql, detector_selection_sql]
-        .filter(_ => _)
+        .filter((_) => _)
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply app/lib/database/views/flags_view.js:45]: right

[the remark this answers is not in the document]
Comment (xsalonx) [reply app/lib/database/views/flags_view.js]: refactored

File: app/lib/database/utilities/PGQueryBuilder.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (xsalonx) [reply app/lib/database/utilities/PGQueryBuilder.js]: refactored

**Review by martinboulais — APPROVED** [review]

**Pull request merged**